src/main/java/dao/GenDao.java

```diff
@@ -0,0 +1,131 @@
+package dao;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class GenDao<T> {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    protected SessionFactory getSessionFactory() {
+        return sessionFactory;
+    }
+
+    protected final boolean addObject(T object){
+        Session session = sessionFactory.openSession();
+        Transaction tx=null;
+        try{
+            tx = session.getTransaction();
+            tx.begin();
+            session.save(object);
+            tx.commit();
+
+        }catch(Exception e){
+            if(tx!=null)
+                tx.rollback();
+            return false;
+        }
+        finally {
+            session.close();
+        }
+        return true;
+    }
+
+    protected final boolean removeObject(T object){
```

Comment:
jeśli już zrobiłes takie metody to czy nie dałoby się czegoś z nich wyciągnąć przed nawias? wyglądają one praktycznie tak samo i różnią się na oko jedną linią